docs/examples/machine_translation/transformer.md

```diff
@@ -132,22 +126,28 @@ wmt_test_batchify_fn = nlp.data.batchify.Tuple(
     nlp.data.batchify.Stack())
 ```
 
-We can then construct bucketing samplers, which generate batches by grouping sequences with similar lengths.
+In GluonNLP, all dataset items are tuples. In the preprocessed `wmt_data_test_with_len`, it includes
+`(src, tgt, len(src), len(tgt), idx)` elements. In order to express how we'd like to batchify them
+in GluonNLP, we use the built-in batchify functions.
+
+* [Tuple](../../api/data.batchify.rst#gluonnlp.data.batchify.Tuple) is the GluonNLP way of applying different batchify functions to each element of a dataset item. In this case, we are applying `Pad` to `src` and `tgt`, `Stack` to `len(src)` and `len(tgt)` with conversion to float32, and simple `Stack` to `idx` without type conversion.
+* [Pad](../../api/data.batchify.rst#gluonnlp.data.batchify.Pad) takes the elements from all dataset items in a batch, and pad them according to the item of maximum length to form a padded matrix/tensor.
+* [Stack](../../api/data.batchify.rst#gluonnlp.data.batchify.Stack) simply stacks all elements in a batch, and requires all elements to be of the same length.
 
-```{.python .input}
-wmt_bucket_scheme = nlp.data.ExpWidthBucket(bucket_len_step=1.2)
-```
+
+We can then construct bucketing samplers, which generate batches by grouping sequences with similar lengths. Here, we use [FixedBucketSampler](../../api/data.rst#gluonnlp.data.FixedBucketSampler) with [ExpWidthBucket](../../api/data.rst#gluonnlp.data.ExpWidthBucket). With this setting, the sampler would select buckets following an approximately exponentially increasing interval of maximum bucket lengths.
```

Comment:
We could explain the bucketsampler a little bit. Insert this sentence `FixedBucketSampler aims to assign each data sample to a fixed bucket based on its length.` before `With this setting, the sampler would select buckets following an approximately exponentially increasing interval of maximum bucket lengths.`